File: lesson9/domain/src/main/java/ru/mirea/vasilev/domain/usecases/GetFavoriteFilmUseCase.java
package ru.mirea.vasilev.domain.usecases;

import ru.mirea.vasilev.domain.models.Movie;
import ru.mirea.vasilev.domain.repository.MovieRepository;

public class GetFavoriteFilmUseCase {
    private MovieRepository movieRepository;

    public GetFavoriteFilmUseCase(MovieRepository movieRepository) {
        this.movieRepository = movieRepository;
    }

    public Movie execute() {
        return movieRepository.getMovie();
    }
}

File: lesson9/app/src/main/java/ru/mirea/vasilevmn/lesson9/domain/usecases/GetFavoriteFilmUseCase.kt
package ru.mirea.vasilevmn.lesson9.domain.usecases

import ru.mirea.vasilevmn.lesson9.domain.models.Movie
import ru.mirea.vasilevmn.lesson9.domain.repository.MovieRepository


class GetFavoriteFilmUseCase(private val movieRepository: MovieRepository) {
    fun execute(): Movie? {
        return movieRepository.movie
    }
}